quipucords/scanner/network/utils.py

```diff
@@ -173,3 +181,54 @@ def get_fact_names():
 def raw_facts_template():
     """Results template for fact collection on network scans."""
     return {fact_name: None for fact_name in get_fact_names()}
+
+
+GEN_KEYFILES_DIRECTORY_PREFIX = "quipucords_ssh_private_key_files_"
+GEN_KEYFILE_PREFIX = "ssh_private_key_file_credential_"
+
+
+def generate_ssh_keyfile(credential, ssh_keyvalue):
+    """Generate an ssh private keyfile for the ssh key content specified."""
+    private_keyfile_path = os.path.join(
+        tempfile.mkdtemp(prefix=GEN_KEYFILES_DIRECTORY_PREFIX),
+        f"{GEN_KEYFILE_PREFIX}{credential.get('id')}",
+    )
+    private_key_fp = open(private_keyfile_path, "w+")
+    private_key_fp.write(ssh_keyvalue)
+    private_key_fp.write("\n")
+    private_key_fp.close()
```

Comment:
Also, whether we use `tempfile` or not, at least these lines should use the `open` context manager to better guarantee that `close` is always automatically called. Like
```suggestion
    with open(private_keyfile_path, "w+") as private_key_fp:
      private_key_fp.write(ssh_keyvalue)
      private_key_fp.write("\n")
```